### PR TITLE
feat: refactor fake backend onto new API (RUN-38193)

### PR DESCRIPTION
**Pull request by eliranw**
## Summary
- Switch status-updater from `ClusterTopology` to `ClusterConfig` internally
- Add `ResolveNodePool` function that handles profile-based resolution (Load→Merge→Extract) and backwards-compat path (direct overrides from normalization)
- All downstream components (device-plugin, kwok-*, topology-server, status-exporter) unchanged — they consume `NodeTopology` CMs which remain the same
- 5 new unit tests for `ResolveNodePool` covering: profile resolution, profile+overrides, backwards-compat, resources, and missing profile error

**Depends on:** Phase 1 (#183) and Phase 2 (#184)

## Test plan
- [x] All existing unit tests pass (48+ tests)
- [x] `ResolveNodePool` tests cover both profile and override paths
- [x] Full build compiles clean
- [x] E2E: deploy with old `topology:` format → verify identical behavior
- [x] E2E: deploy with new `cluster:` format + `profile: h100` → verify profile resolution

### CODE REVIEW

File: internal/common/topology/resolve.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	resolved.GpuCount = spec.GpuCount
+
+	return resolved, nil
+}
+
+// resolveFromOverrides extracts GPU spec directly from overrides.
+// This is the backwards-compatible path used when normalization converts
+// old-format fields (gpuProduct, gpuMemory, gpuCount) into overrides.
+func resolveFromOverrides(pool NodePoolConfig, resolved *ResolvedPool) (*ResolvedPool, error) {
+	overrides := pool.Gpu.Overrides
```
Comment (iris-shain-runai) [inline internal/common/topology/resolve.go:68]: looks like error is always nil, right?

**Review by iris-shain-runai — APPROVED** [review]

**Comment by iris-shain-runai** [comment]
Doing a second pass now that #183/#184 have landed — approval stands, but flagging a couple of things worth addressing before merge and some smaller follow-ups.

### Worth addressing before merge

1. **`sanitizeLabelValue` is incomplete** — `internal/status-exporter/export/labels/labels_builder.go` only replaces spaces. K8s label values must match `[a-zA-Z0-9_.-]{0,63}`. A GPU product with `/`, `(`, `)`, etc. will fail validation and the node update errors out. There's also no 63-char cap. Consider aligning with NVIDIA GFD's sanitizer (regex replace + truncate), and add a unit test covering the edge cases (invalid chars, overflow length, empty).

2. **Non-deterministic `OtherDevices` order** — in `ResolveNodePool` (`internal/common/topology/resolve.go`), iterating `pool.Resources` (`[]map[string]int`) is random for the inner map, so the resulting slice order flaps between reconciles. If anything downstream is order-sensitive (CM diff, label generation), this causes churn. Sort by name before returning.

### Follow-ups / nits

3. **`DeviceCount` treats explicit `0` as "unset"** (`internal/common/profile/profile.go`). `device_count: 0` silently falls back to `len(devices)`. Probably fine in practice, but worth a comment so future-us doesn't chase a ghost.

4. **Breaking label-key rename** `run.ai/gpu-profile` → `fake-gpu-operator/gpu-profile` isn't in the PR body. Anyone with an external selector relying on the old key would break silently — worth calling out in release notes.

5. **`release-helm` on `workflow_dispatch`** — `release_tag` is `required: true`, so `inputs.release_tag == ''` is always false and helm is always skipped on manual dispatch. Looks intentional ("image-only release") but a one-line comment above the `if:` would help future readers.

6. **CI job duplication** — `e2e` and `e2e-profiles` have identical setup (`.github/workflows/ci.yml`). A `strategy.matrix` with `{format: [topology, profiles]}` would cut the file in half and parallelize cleanly — kind-cluster start is most of the wall-clock.

7. **Minor**:
   - `expectedGpuProduct` / `expectedGpuCount` / `expectedHighend*` in `test/e2e/e2e_test.go` can be `const`.
   - `topology_cm.go` reads the namespace from viper each call — injecting once in `NewNodeHandler` is cleaner and easier to test.
   - Integer truncation in `resolveFromOverrides` memory path (`total_bytes / (1024*1024)`) — safe for realistic sizes, but worth a comment.

Nice refactor overall. The backwards-compat path being explicitly tested (`TestResolveNodePool_FromOverrides_BackwardsCompat`) is great, and pinning both values files to the same expected product/count makes the `e2e` vs `e2e-profiles` comparison meaningful.
